src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -170,12 +170,39 @@ public void TriageDataLocks(Party party, List<long> dataLockEventIds, TriageStat
             ChangeTrackingSession.CompleteTrackingSession();
         }
 
-        public void ReplacePriceHistory(Party party, List<PriceHistory> updatedPriceHistory, UserInfo userInfo)
+        public void ReplacePriceHistory(Party party, List<PriceHistory> currentPriceHistory, List<PriceHistory> updatedPriceHistory, UserInfo userInfo)
         {
-            StartTrackingSession(UserAction.UpdatePriceHistory, party, Cohort.EmployerAccountId, Cohort.ProviderId, userInfo);
-            this.PriceHistory = updatedPriceHistory;
-            ChangeTrackingSession.TrackUpdate(this);
+            StartTrackingSession(UserAction.DeletePriceHistory, party, Cohort.EmployerAccountId, Cohort.ProviderId, userInfo, Id);
+            foreach (var priceHistory in currentPriceHistory)
+            {
+                if (updatedPriceHistory.All(x => x.Cost != priceHistory.Cost))
+                {
+                    ChangeTrackingSession.TrackDelete(priceHistory);
+                }
+            }
+            ChangeTrackingSession.CompleteTrackingSession();
+
+            StartTrackingSession(UserAction.UpdatePriceHistory, party, Cohort.EmployerAccountId, Cohort.ProviderId, userInfo, Id);
```

Comment:
In the tracking session, the user action should probably just be `TriageDataLock`

then can remove `InsertDataLock` and `DeleteDataLock` options from the enum